buildSrc/src/test/java/org/opensearch/gradle/pluginzip/PublishTests.java

```diff
@@ -50,156 +56,410 @@ public void tearDown() {
+        // not allow for it ATM. If, however, it is fixed in the future the following is the code that can
+        // be used...
+
+        Project project = ProjectBuilder.builder().build();
+        project.getPluginManager().apply(Publish.class);
+        // add publications via API
+
+        // evaluate the project
+        ((DefaultProject)project).evaluate();
+
+        // - Check that "validatePluginZipPom" and/or "publishPluginZipPublicationToZipStagingRepository"
+        //   tasks have dependencies on "generatePomFileForNebulaPublication".
+        // - Check that there is the staging repository added.
+
+        // However, due to known issue(1): https://github.com/gradle/gradle/issues/20301
+        // it is impossible to reach to individual tasks and work with them.
+        // (1): https://docs.gradle.org/7.4/release-notes.html#known-issues
+
+        // I.e.: The following code throws exception, basically any access to individual tasks fails.
+        project.getTasks().getByName("validatePluginZipPom");
+         ------------------------------- */
+
+        // Instead, we run the gradle project via GradleRunner (this way we get fully evaluated project)
+        // and using the minimal possible configuration (missingPOMEntity) we test that as soon as the zip publication
+        // configuration is specified then all the necessary tasks are hooked up and executed correctly.
+        // However, this does not test execution order of the tasks.
+        GradleRunner runner = prepareGradleRunnerFromTemplate("missingPOMEntity.gradle", ZIP_PUBLISH_TASK/*, "-m"*/);
         BuildResult result = runner.build();
-        // Check if task publishMavenzipPublicationToZipstagingRepository has ran well
-        assertEquals(SUCCESS, result.task(":" + zipPublishTask).getOutcome());
-        // check if the zip has been published to local staging repo
+
+        assertEquals(SUCCESS, result.task(":" + "bundlePlugin").getOutcome());
+        assertEquals(SUCCESS, result.task(":" + "generatePomFileForNebulaPublication").getOutcome());
+        assertEquals(SUCCESS, result.task(":" + "generatePomFileForPluginZipPublication").getOutcome());
+        assertEquals(SUCCESS, result.task(":" + ZIP_PUBLISH_TASK).getOutcome());
+    }
+
+    /**
+     * If the plugin is used but relevant publication is not defined then a message is printed.
+     */
+    @Test
+    public void missingPublications() throws IOException, URISyntaxException {
+        GradleRunner runner = prepareGradleRunnerFromTemplate("missingPublications.gradle", "build", "-m");
+        BuildResult result = runner.build();
+
+        assertTrue(result.getOutput().contains("Plugin 'opensearch.pluginzip' is applied but no 'pluginZip' publication is defined."));
+    }
+
+    @Test
+    public void missingGroupValue() throws IOException, URISyntaxException, XmlPullParserException {
```

Comment:
@lukas-vlcek this is one of the points we have talked before, if `group` is missing we should default to `org.opensearch.plugin` in order to not break the existing plugins - since this is `2.x` and we do not want breaking changes there

Reply:
Ah! Great point, I completely forgot about this :-) I will push update.

Reply:
Fixed.